scripts/deploy-on-kind.sh

```diff
@@ -62,15 +59,15 @@ ensure_local_registry() {
 [plugins."io.containerd.grpc.v1.cri".registry]
   [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
     [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localregistry-docker-registry.default.svc.cluster.local:30050"]
-      endpoint = ["http://localregistry-docker-registry.default.svc.cluster.local:30050"]
+      endpoint = ["http://127.0.0.1:30050"]
```

Comment:
Out of curiosity how does the networking work for this? Maybe I just need to read up more on KinD networking.. 😅

Reply:
1. We set the port to 30050 so that we can map it to a node port with the same port number
2. We tell the helm chart that we want a nodeport with 30050
3. We tell containerd to mirror the k8s dns for the service to localhost (we used to just put the hostname in /etc/hosts but this PR makes that unnecessary)
4. we tell everybody to use localregistry-docker-registry.default.svc.cluster.local:30050. For the build service, kube dns resolves the address. For k8s itself, the containerd config above redirects it to the nodeport we opened.

This approach assumes that we will just have a single kind worker node, but it could be elaborated if we wanted to use multi-node clusters.